### PR TITLE
BQ lineage crawler needs to filter out duplicated [sc-6440]

### PR DESCRIPTION
**Pull request by alyiwang**
### 🤔 Why?

the referenced tables in big query log may contain duplicates

### 🤓 What?

- remove duplicates and self-referencing when parsing a log

### 🧪 Tested?

unit tests

### CODE REVIEW

File: tests/bigquery/lineage/data/sample_log.json [hunk 1]
```diff
@@ -118,6 +118,8 @@
                 "billingTier": 1,
```
Comment (mars-lan) [inline tests/bigquery/lineage/data/sample_log.json:118]: Please also add a test for self reference.

Comment (alyiwang) [reply tests/bigquery/lineage/data/sample_log.json:118]: There is self reference, the test3 is the destination dataset

Comment (mars-lan) [reply tests/bigquery/lineage/data/sample_log.json:118]: Argh I see. Duplicated self reference :)

**Review by mars-lan — APPROVED** [review]

File: tests/bigquery/lineage/data/sample_log.json (hunk 1 above)
Comment (alyiwang) [reply tests/bigquery/lineage/data/sample_log.json:118]: test2 is duplicate, test3 is self-reference table

**Pull request merged**